internal/mesh/internal/controllers/routes/generate.go

```diff
@@ -329,6 +328,58 @@ func compile(
 	}
 }
 
+func compileFailoverConfig(
+	related *loader.RelatedResources,
+	failoverConfig *pbcatalog.FailoverConfig,
+	targets map[string]*pbmesh.BackendTargetDetails,
+) *pbmesh.ComputedFailoverConfig {
+	if failoverConfig == nil {
+		return nil
+	}
+
+	cfc := &pbmesh.ComputedFailoverConfig{
+		Destinations:  make([]*pbmesh.ComputedFailoverDestination, 0, len(failoverConfig.Destinations)),
+		Mode:          failoverConfig.Mode,
+		Regions:       failoverConfig.Regions,
+		SamenessGroup: failoverConfig.SamenessGroup,
+	}
+
+	for _, dest := range failoverConfig.Destinations {
+		backendRef := &pbmesh.BackendReference{
+			Ref:        dest.Ref,
+			Port:       dest.Port,
+			Datacenter: dest.Datacenter,
+		}
+
+		destSvcRef := dest.Ref
+
+		svc := related.GetService(destSvcRef)
+
+		var backendTargetName string
+		ok, meshPort := shouldRouteTrafficToBackend(svc, backendRef)
+		if !ok {
+			continue // skip this leg of failover for now
```

Comment:
The alternative here would be to create a failover destination of the "null route cluster" which conceptually doesn't make sense as failover is supposed to _increase_ uptime.